facebook/OpenBIC
# fby3.5: rf: Support fru

Summary:
- Support read and write FRU.

Dependency: #300 

Test plan:
- Build code: Pass
- Read FRU: Pass
- Write FRU: Pass

Log:
1. Check RF FRU can read and write when FRU is empty.
(1) Read failed due to FRU is empty
root@bmc-oob:~# fruid-util slot2 1U
Failed print FRUID for Server board 2 1U
Check syslog for errors! (err 2h)

(2) Write test FRU data
root@bmc-oob:~# fruid-util slot2 1U --write /tmp/fruid_1ou_test.bin

(3) Read FRU again
root@bmc-oob:~# rm -rf /tmp/fruid_slot2_dev19.bin
root@bmc-oob:~# /usr/local/bin/bic-cached -f slot2
root@bmc-oob:~# fruid-util slot2 1U

FRU Information           : Server board 2 1U
---------------           : ------------------
Board Mfg Date            : Mon Jul 12 04:51:00 2021
Board Mfg                 : test
Board Product             : test
Board Serial              : test
Board Part Number         : test
Board FRU ID              : test
Product Manufacturer      : Mellanox
Product Name              : Mellanox ConnectX-6 DX OCP3.0
Product Part Number       : MCX623435MC-CDAE_FB
Product Version           : A8
Product Serial            : MT2129T00085
Product Custom Data 3     : ConnectX-6 DX

2. Check RF FRU can read when FRU is normal.
root@bmc-oob:~# fruid-util slot2 1U

FRU Information           : Server board 2 1U
---------------           : ------------------
Board Mfg Date            : Mon Jul 12 04:51:00 2021
Board Mfg                 : test
Board Product             : test
Board Serial              : test
Board Part Number         : test
Board FRU ID              : test
Product Manufacturer      : Mellanox
Product Name              : Mellanox ConnectX-6 DX OCP3.0
Product Part Number       : MCX623435MC-CDAE_FB
Product Version           : A8
Product Serial            : MT2129T00085
Product Custom Data 3     : ConnectX-6 DX
